src/openrct2-ui/windows/StaffList.cpp

```diff
@@ -531,6 +532,30 @@ class StaffListWindow final : public Window
             auto intent = Intent(WindowClass::Peep);
             intent.PutExtra(INTENT_EXTRA_PEEP, staff);
             ContextOpenIntent(&intent);
+
+            // If autoposition of staff is disabled, set the pickup tool on the newly created staff window
+            if (staff->State == PeepState::Picked)
+            {
+                picked_peep_old_x = staff->x;
+                CoordsXYZ nullLoc{};
+                nullLoc.SetNull();
+
+                PeepPickupAction pickupAction{ PeepPickupType::Pickup,
+                                               EntityId::FromUnderlying(staff->sprite_index.ToUnderlying()), nullLoc,
+                                               NetworkGetCurrentPlayerId() };
```

Comment:
On L525 you could give `const GameAction*` a name and then use the game actions `GetPlayer()` function. It's all going to be the same value but it feels neater in my mind.

Reply:
These two methods don't return the same value in my testing. In single player, `GetPlayer()` returns `-1` and `NetworkGetCurrentPlayerId()` returns `0` and so the pickup action doesn't work correctly. Am I missing something?

Reply:
Oh, that's a bit surprising. Leave as is then